### PR TITLE
refactor: remove unmaintained surf for reqwest

### PR DESCRIPTION
**Pull request by KolbyML**
### What was wrong?
fixes https://github.com/ethereum/trin/issues/1446 surf has been unmaintained for over 2 years now, it is also using a old version of packages which prevents us from compiling Trin to Risc-v https://github.com/ethereum/trin/issues/1444
### How was it fixed?

Just convert `surf` -> `reqwest` equivalents, which was pretty straight forward

### CODE REVIEW

File: src/bin/test_providers.rs
Comment (morph-dev) [inline src/bin/test_providers.rs:438]: nit: I think this can be simplified to something like:

```rust
let response = Client::new()
    .post("...")
    .headers(headers)
    .json(&request)
    .send()
    .await
    .map_err(|e| ...)?
```

File: src/bin/test_providers.rs
Comment (morph-dev) [inline src/bin/test_providers.rs:455]: nit: we could use `.json::<serde_json::Value>()`

File: trin-execution/src/era/manager.rs
Comment (morph-dev) [inline trin-execution/src/era/manager.rs:36]: Now that I think about it, it doesn't make sense for `CONTENT_TYPE` header to be set here and passed to `get_era_files` and `get_era1_files` (those functions are in a different crate, they control the URL and they are responsible for parsing the response).

So in this case, I would probably just create simple `Client::new()`, and each function would initialize headers that they need.

Not really part of this pr, but I think it's small enough to be included.

**Review by morph-dev — COMMENTED** [review]
Mostly looks good, but there is one big issue in `portal-bridge` (see comment).

Also, `light-client` crate also depends on reqwest, so we should update that dependency (should use workspace version)

File: portal-bridge/src/api/execution.rs
Comment (njgheorghita) [inline portal-bridge/src/api/execution.rs:57]: Comment on line 276 should be removed, and probably some testing to make sure that `reqwest` can handle the async load, though it's a very popular library so I expect it will

File: portal-bridge/src/cli.rs
Comment (KolbyML) [reply portal-bridge/src/cli.rs:157]: I will just implement base_url for Reqwest

**Comment by KolbyML** [comment]
@njgheorghita @morph-dev ready for another review. I believe I resolved all the PR concerns
image
image
Here are a few pictures of history latest nothing seems wrong

Running backfill using infura I see what you mean nick when you said shisui uTP failures. It looks good as well

**Comment by njgheorghita** [comment]
Cool. I'm guessing this comment predates era1 files, so back then when in `backfill` mode, we'd be making quite a bit of parallel http requests. But, now that we have era1 files (and plan to use some version of era files for everything from merge -> latest?) I actually don't think this is much of an issue anymore. Unless there's somewhere else in the bridge that I'm missing where we'll have large amounts of parallel http requests....

**Comment by KolbyML** [comment]
> large amounts of parallel http requests

I just ran backfill using infura as I said before, so it was making a large amount of parallel http requests. Reqwest's get 100 times more downloads a month than surf and is also built on hyper so I don't see why it wouldn't be able to handle `large amounts of parallel http requests`

**Review by morph-dev — APPROVED** [review]
LGTM. But wait for Nick's approval as well.

**Review by ogenev — APPROVED** [review]
LGTM

**Review by njgheorghita — APPROVED** [review]

**Pull request merged**
